### PR TITLE
Virtual Limbsanity

### LINKED ISSUE #66251: Invisible icons when repainting augments
## Round ID:

local server running latest tgcode

## Testmerges:

none

## Reproduction:

try repainting a cyborg limb with a spraycan right click. following happens:
image

### PR DESCRIPTION
**Pull request by Kapu1178**
https://www.youtube.com/watch?v=4JkIs37a2JE

## About The Pull Request
Fixes #66251

General code cleanup, adds a new footgun I pray people don't misuse. Oh, and adds a new unit test to ensure all bodypart appearances are valid by default.

## Changelog

:cl:
fix: Robotic limbs are once again style-able using a spraycan.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
what footgun

**Comment by Kapu1178** [comment]
> what footgun

`/obj/item/bodypart/proc/change_appearance()`

**Comment by Kapu1178** [comment]
Why do i not get merge conflict notifs

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 1]
```diff
@@ -1084,3 +1047,44 @@
 /obj/item/bodypart/proc/recolor_external_organs()
 	for(var/obj/item/organ/external/ext_organ as anything in external_organs)
 		ext_organ.inherit_color(force = TRUE)
+
+///A multi-purpose setter for all things immediately important to the icon and iconstate of the limb.
+/obj/item/bodypart/proc/change_appearance(icon, id, greyscale, dimorphic)
```
Comment (Kylerace) [inline code/modules/surgery/bodyparts/_bodyparts.dm:1052]: should you be able to change a limb from dimorphic to not?

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/surgery/bodyparts/_bodyparts.dm (hunk 1 above)
Comment (Kapu1178) [reply code/modules/surgery/bodyparts/_bodyparts.dm:1052]: > should you be able to change a limb from dimorphic to not?

You shouldn't be allowed to do most of these things, but for the sake of giving people full control over how limbs appear, its essential.

Say you wanted a robotic chest to appear as a human chest. You need the ability to change it's dimorphism to TRUE to ensure booba shows up on wamen.

**Pull request merged**